kotlinpoet/src/main/java/com/squareup/kotlinpoet/FileSpec.kt

```diff
@@ -147,14 +148,26 @@ public class FileSpec private constructor(
 
     val importedTypeNames = codeWriter.importedTypes.values.map { it.canonicalName }
     val importedMemberNames = codeWriter.importedMembers.values.map { it.canonicalName }
+
+    // If we don't have default imports or are collecting them, we don't need to filter
+    var defaultImportsFilter: (String) -> Boolean = { false }
+    if (!collectingImports && defaultImports.isNotEmpty()) {
+      val defaultImports = defaultImports.map(String::escapeSegmentsIfNecessary)
+      defaultImportsFilter = { importName ->
+        importName.substringBeforeLast(".") in defaultImports
+      }
+    }
     // Aliased imports should always appear at the bottom of the imports list.
     val (aliasedImports, nonAliasedImports) = memberImports.values.partition { it.alias != null }
     val imports = (importedTypeNames + importedMemberNames)
+      .asSequence()
       .filterNot { it in memberImports.keys }
       .map { it.escapeSegmentsIfNecessary() }
-      .plus(nonAliasedImports.map { it.toString() })
+      .plus(nonAliasedImports.asSequence().map { it.toString() })
+      .filterNot(defaultImportsFilter)
       .toSortedSet()
       .plus(aliasedImports.map { it.toString() }.toSortedSet())
+      .toList()
```

Comment:
nit: Is `toList()` needed here, given that there's a `toSortedSet()` above?

Reply:
Good point, I'll check. This was automatically added by IJ when converting to a sequence

Reply:
1434cac